nexus/db-queries/benches/harness/mod.rs

```diff
@@ -0,0 +1,202 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Shared Test Harness for benchmarking database queries
+//!
+//! This structure shares logic between benchmarks, making it easy
+//! to perform shared tasks such as creating contention for reservations.
+
+use nexus_db_queries::authz;
+use nexus_db_queries::context::OpContext;
+use nexus_db_queries::db::pub_test_utils::TestDatabase;
+use nexus_db_queries::db::DataStore;
+use nexus_test_utils::sql::process_rows;
+use nexus_test_utils::sql::Row;
+use omicron_test_utils::dev;
+use slog::Logger;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+pub(crate) mod db_utils;
+
+use db_utils::*;
+
+pub struct TestHarness {
+    db: TestDatabase,
+    authz_project: authz::Project,
+}
+
+struct ContentionQuery {
+    sql: &'static str,
+    description: &'static str,
+}
+
+const QUERIES: [ContentionQuery; 4] = [
+    ContentionQuery {
+        sql: "SELECT table_name, index_name, num_contention_events::TEXT FROM crdb_internal.cluster_contended_indexes",
+        description: "Indexes which are experiencing contention",
+    },
+    ContentionQuery {
+        sql: "SELECT table_name,num_contention_events::TEXT FROM crdb_internal.cluster_contended_tables",
+        description: "Tables which are experiencing contention",
+    },
+    ContentionQuery {
+        sql: "WITH c AS (SELECT DISTINCT ON (table_id, index_id) table_id, index_id, num_contention_events AS events, cumulative_contention_time AS time FROM crdb_internal.cluster_contention_events) SELECT i.descriptor_name as table_name, i.index_name, c.events::TEXT, c.time::TEXT FROM crdb_internal.table_indexes AS i JOIN c ON i.descriptor_id = c.table_id AND i.index_id = c.index_id ORDER BY c.time DESC LIMIT 10;",
+        description: "Top ten longest contention events, grouped by table + index",
+    },
```

Comment:
sorry to be annoying, but...i don't suppose we could wrap some of these queries?

Reply:
Updated!